src/renderer/component/selectChannel/view.jsx

```diff
@@ -0,0 +1,221 @@
+    const { onChannelChange } = this.props;
+    const channel = event.target.value;
+
+    if (channel === 'new') {
+      this.setState({ addingChannel: true });
+      onChannelChange(channel);
+    } else {
+      this.setState({ addingChannel: false });
+      onChannelChange(channel);
+    }
+  }
+
+  handleNewChannelNameChange(event: SyntheticInputEvent<*>) {
+    let newChannelName = event.target.value;
+
+    if (newChannelName.startsWith("@")) {
+      newChannelName = newChannelName.slice(1);
+    }
+
+    let newChannelNameError;
+    if (newChannelName.length > 1 && !isNameValid(newChannelName.substr(1), false)) {
+      newChannelNameError = __('LBRY channel names must contain only letters, numbers and dashes.');
+    }
+
+    this.setState({
+      newChannelNameError,
+      newChannelName,
+    });
+  }
+
+  handleNewChannelBidChange(event: SyntheticInputEvent<*>) {
+    const { balance } = this.props;
+    const newChannelBid = parseFloat(event.target.value);
+    this.setState({
+      newChannelBid,
+      newChannelBidError: newChannelBid > balance ? __("Not enough credits") : ""
+    });
+  }
+
+  handleCreateChannelClick() {
+    const { balance, createChannel, onChannelChange } = this.props;
+    const { newChannelBid, newChannelName } = this.state;
+
+    let channelName = `@${newChannelName}`;
+
+    if (newChannelBid > balance) {
+      return;
+    }
+
+    this.setState({
+      creatingChannel: true,
+      createChannelError: undefined,
+    });
+
+
+    const success = () => {
+      this.setState({
+        creatingChannel: false,
+        addingChannel: false,
+      });
+
+      onChannelChange(channelName);
+    };
+
+    const failure = err => {
+      this.setState({
+        creatingChannel: false,
+        createChannelError: __('Unable to create channel due to an internal error.')
+      });
+    };
+
+
+    createChannel(channelName, newChannelBid).then(success, failure);
+  }
+
+  render() {
+    const channel = this.state.addingChannel ? 'new' : this.props.channel;
+    const { fetchingChannels, channels = [] } = this.props;
+    const {
+      newChannelName,
+      newChannelNameError,
+      newChannelBid,
+      newChannelBidError,
+      creatingChannel,
+      createChannelError,
+      addingChannel
+    } = this.state;
+
+    return (
+        <div className="card__content">
+          {createChannelError && (
+            <div className="form-field__error">{createChannelError}</div>
+          )}
+          {fetchingChannels ? (
+            <BusyMessage message="Updating channels" />
+          ) : (
+              <FormField
+                key="channel"
+                type="select"
+                tabIndex="1"
+                onChange={this.handleChannelChange}
+                value={channel}
+              >
+                <option key="anonymous" value="anonymous">
+                  {__('Anonymous')}
+                </option>
+                {channels.map(({ name }) => (
```

Comment:
It'd be cool if this defaulted to the last channel you published with.

Reply:
That's a good idea